src/pipx/commands/inject.py

```diff
@@ -95,8 +96,25 @@ def inject(
         raise PipxError(
             "Cannot pass --include-deps if --include-apps is not passed as well"
         )
+    if not package_specs and not requirement_files:
+        raise PipxError(
+            "Dependencies or requirements files must be provided. See `pipx inject -h` for more info"
+        )
+
+    if package_specs and requirement_files:
+        raise PipxError(
+            "Dependencies and --requirement/-r option cannot be passed at the same time"
+        )
+
+    if requirement_files:
+        packages_list = []
+        for file in requirement_files:
+            with open(Path(file), "r") as f:
```

Comment:
Instead of reading the files, should we just forward the path to pip to let it parse instead? Requirements files are technically a pip-specific format and parsing them ourselves seems like a big future source of code bloat down the road. Instead we should just redirect all the responsibilities to pip.

Reply:
But the question is that how should we determine package names?

Reply:
Why do we need the package names? This is `inject` not `install`.

Reply:
Since we need to forward those package names to `pipx_metadata`.

Reply:
Would it be viable to do a `pip freeze` before and after installation and record the diff?

Reply:
Oh, or we can ask pip to provide an installation report! We can simply record packages with the `requested` flag. https://pip.pypa.io/en/stable/reference/installation-report/

Reply:
Haven't heard about this before. I will check it out and see if it works.

Reply:
I just took a look at the `pipx_metadata` and turns out `pip_args` can only be passed once currently. They will be forwarded to each `pip install` command during the injection. Perhaps we will need to make some changes to the metadata format if we want to pass the requirements file to pip directly, since there might be different options provided for each package.

Reply:
> Oh, or we can ask pip to provide an installation report! We can simply record packages with the requested flag. **[pip.pypa.io/en/stable/reference/installation-report](https://pip.pypa.io/en/stable/reference/installation-report/)**

But it looks like the installation report will also report the dependencies of a package, which are not specified in requirements.txt. Though we can exclude them from pipx_metadata by checking whether they are in the `requires_dist` field, I personally think this is too complicated. 

Here's my own solution that should solve this issue:
- Parse the requirements file line by line
- Use `split()` to split those arguments out
- Check if the argument starts with `-` or `--`, which means they are options and should be added to `pip_args`
- Have the installation as usual
- Repeat the above steps till the last line

Any thoughts?

Reply:
> But it looks like the installation report will also report the dependencies of a package, which are not specified in requirements.txt. Though we can exclude them from pipx_metadata by checking whether they are in the `requires_dist` field, I personally think this is too complicated

There’s a field `requested` for this, we can simply filter out entries with this set to true (meaning it’s from user input i.e. the requirements file).

Reply:
Oops, didn't notice that, thanks for pointing it out!